src/Parsing.cc

```diff
@@ -142,21 +121,15 @@ int
 GetInteger(void)
 {
     char *token = ConfigParser::NextToken();
-    int i;
-
-    if (!token) {
-        self_destruct();
-        return -1; // not reachable
-    }
+    if (!token)
+        throw Cfg::FatalError("number is missing");
 
     // The conversion must honor 0 and 0x prefixes, which are important for things like umask
     int64_t ret = xatoll(token, 0);
 
-    i = (int) ret;
-    if (ret != static_cast<int64_t>(i)) {
-        debugs(0, DBG_PARSE_NOTE(DBG_IMPORTANT), "ERROR: The value '" << token << "' is larger than the type 'int'.");
-        self_destruct();
-    }
+    int i = (int) ret;
+    if (ret != static_cast<int64_t>(i))
+        throw Cfg::FatalError(ToSBuf("invalid value: '", token, "' is larger than the type 'int'"));
```

Comment:
```suggestion
        throw Cfg::FatalError(ToSBuf("invalid value: '", token, "' does not fit in a 64-bit signed integer"));
```

Reply:
Proposed change would be incorrect. This is a check that the value has not overflowed the size of "int" when being reinterpreted.

The value clearly does fit into a *64-bit* since xatoll() provided it to us as one. This function is trying to down-size it into an "int" type for the caller.

Reply:
Got you.